### ISSUE TITLE
Need to refactor SrchSrv

### ISSUE THREAD

**Issue by iamnemesis** [pritamroy11/multi-doc-summ#1]
-Separation of concerns
-Clean code
-Use new Java features

**Comment by SauravDas90** [comment]
@iamnemesis what you suggest ? use a framework for this project?

**Comment by iamnemesis** [comment]
@SauravDas90 @pritamroy11 framework will be a BIG overkill...better to just refactor.

**Comment by pritamroy11** [comment]
yes, if we want to use a new framework, it will be a lot of work plus architecture design. we can clean the code and do some optimization and make it better.
And I have given you guys full access.

**Comment by SauravDas90** [comment]
For code Structure
1. Use JPA for querying
2. add a seperate table for moving validation for admin login @iamnemesis 
3. encryption and decryption needed for user login
4. Db structure needs to be provided and updated @iamnemesis 
5. For validation we can move code from jsp to javascript, angular/backbone(preferred) @pritamroy11

**Comment by pritamroy11** [comment]
1. JPA is a interface type of concept. Then we have to use hibernate or something to implement according to JPA and have to create entity POJO classes for each database table. OK, I will do that.
2. and 3. can be achieved easily.
3. have to create database.sql with proper table name and attributes @iamnemesis @SauravDas90 
4. if we want to use angular/backbone then, we have to re-structure the whole application. I don't have that much time. We will decide it later on.

**Comment by SauravDas90** [comment]
1 should be done , because plain JDBC is not a good idea
3. Once I have the initial db I can work and improve the idea @iamnemesis
4.For UI I can take charge but need css help, also we need to evaluate if at this stage going for angular/backbone is worth it @pritamroy11

**Comment by pritamroy11** [comment]
all is okay apart from number 4. It will take a lot of time bro... If you want to do it by yourself, you are very much welcome!! and for css, I will help you; but you have to first do the architecture using angular/backbone @SauravDas90

**Comment by SauravDas90** [comment]
@pritamroy11 Yes I know, it will be a huge change and agreed , but above all we need to first optimize the algo, once done we can modify the look and feel and move to any UI frameworks. Right now simple jquery/css will do as we dont have huge no of pages.

**Comment by iamnemesis** [comment]
Closing

**Comment by SauravDas90** [comment]
the issue is not resolved yet, how can it be closed @iamnemesis ?

**Comment by iamnemesis** [comment]
Oops...did not mean to close this one @SauravDas90
